llvm/lib/Target/NVPTX/NVPTXInstrInfo.td

```diff
@@ -1150,6 +1150,16 @@ def DoubleConst1 : PatLeaf<(fpimm), [{
   return &N->getValueAPF().getSemantics() == &llvm::APFloat::IEEEdouble() &&
          N->getValueAPF().convertToDouble() == 1.0;
 }]>;
+// Constant -1.0 (double)
+def DoubleConstNeg1 : PatLeaf<(fpimm), [{
+  return &N->getValueAPF().getSemantics() == &llvm::APFloat::IEEEdouble() &&
+         N->getValueAPF().convertToDouble() == -1.0;
+}]>;
+// Constant -1.0 -> 1.0 (double)
```

Comment:
This transform applies to all double constants, not just 1.0/-1.0. You may want to update the comment accordingly.

Also, this defines a node transform, as opposed to node matchers above, so it would be good to separate them a bit visually, so rephrasing the comment would help with that, too.

Reply:
Thanks, resolved it in the latest changes.